### PR TITLE
Feature/813 configurations tests with checkov

### PR DESCRIPTION
**Pull request by trottomv**

### CODE REVIEW

**Review by niccolomineo — DISMISSED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,19 +13,52 @@ permissions:
...
-          python-version: "3.10"
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install -r requirements/test.txt
-      - name: Run Test
-        run: |
-          python3 -m unittest
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
```
Comment (pauloxnet) [inline .github/workflows/test.yml:37]: Upgrade to 3.11

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -13,19 +13,52 @@ permissions:
...
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (pauloxnet) [inline .github/workflows/test.yml:43]: Remove this extra step

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -13,19 +13,52 @@ permissions:
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (pauloxnet) [inline .github/workflows/test.yml:59]: Remove this extra step

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ default_language_version:
   python: python3
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: "v4.2.0"
+    rev: "v4.4.0"
```
Comment (pauloxnet) [inline .pre-commit-config.yaml:5]: Add this changes in a different PR

**Review by pauloxnet — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -13,19 +13,52 @@ permissions:
   contents: read
 
 jobs:
-  build:
+  check-linter:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
```
[the remark this answers is not in the document]
Comment (trottomv) [reply .github/workflows/test.yml:20]: could be upgraded in a separate PR

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -13,19 +13,52 @@ permissions:
...
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
[the remark this answers is not in the document]
Comment (trottomv) [reply .github/workflows/test.yml]: ok

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -13,19 +13,52 @@ permissions:
...
+        python -m pip install -r requirements/test.txt
+    - name: Run Test
+      run: |
+        python3 -m black --check .
+        python3 -m isort --check .
```
[the remark this answers is not in the document]
Comment (trottomv) [reply .github/workflows/test.yml:30]: could be upgrade in a separate PR

File: .github/workflows/test.yml (hunk 1 above)
Comment (trottomv) [reply .github/workflows/test.yml:36]: could be upgraded in a separate PR

File: .github/workflows/test.yml (hunk 2 above)
Comment (trottomv) [reply .github/workflows/test.yml]: ok

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -13,19 +13,52 @@ permissions:
...
+        python -m pip install -r requirements/test.txt
+    - name: Run Check
+      continue-on-error: true
+      run: |
+        checkov --download-external-modules=false --evaluate-variables=true --quiet --directory .
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
```
[the remark this answers is not in the document]
Comment (trottomv) [reply .github/workflows/test.yml:51]: could be upgraded in a separate PR

File: .github/workflows/test.yml (hunk 3 above)
Comment (trottomv) [reply .github/workflows/test.yml]: ok

File: .pre-commit-config.yaml (hunk 4 above)
Comment (trottomv) [reply .pre-commit-config.yaml]: ok

File: .github/workflows/test.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (pauloxnet) [reply .github/workflows/test.yml:20]: ok

**Comment by trottomv** [comment]
consider it a spike for now, postponed

**Pull request closed without merging**